src/java/org/apache/cassandra/db/virtual/AccordVirtualTables.java

```diff
@@ -73,6 +90,88 @@ public DataSet data()
         }
     }
 
+    public static final class CommandStoreCache extends AbstractVirtualTable
+    {
+        private CommandStoreCache(String keyspace)
+        {
+            super(parse(keyspace,
+                        "Accord Command Store Cache Metrics",
+                        "CREATE TABLE accord_command_store_cache(\n" +
+                        "  id int,\n" +
+                        "  queries bigint,\n" +
+                        "  hits bigint,\n" +
+                        "  misses bigint,\n" +
+                        "  PRIMARY KEY (id)" +
+                        ')'));
+        }
+
+        @Override
+        public DataSet data()
+        {
+            AccordService accord = (AccordService) AccordService.instance();
+            SimpleDataSet result = new SimpleDataSet(metadata());
+
+            Map<Integer, AccordStateCache.ImmutableStats> statsByStore = 
+                accord.node().commandStores().map(PreLoadContext.empty(),
+                                                  store -> ((AccordCommandStore) store.commandStore()).cache().stats());
+
+            for (Map.Entry<Integer, AccordStateCache.ImmutableStats> stats : statsByStore.entrySet())
+            {
+                result.row(stats.getKey());
+                result.column("queries", stats.getValue().queries);
+                result.column("hits", stats.getValue().hits);
+                result.column("misses", stats.getValue().misses);
+            }
+
+            return result;
+        }
+    }
+
+    public static final class MigrationStates extends AbstractVirtualTable
+    {
+        private MigrationStates(String keyspace)
+        {
+            super(parse(keyspace,
```

Comment:
how do I know the migration is *done*?

Reply:
Does a range move from "migrating" -> "migrated"?

Reply:
@aweisberg?

Reply:
You know it's done when `migrated_ranges` is the whole ring and should have a single entry. Right now once a migration state has been created for a table it never goes away.